pkg/identity/revocation_db_test.go

```diff
@@ -0,0 +1,153 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package identity_test
+
+import (
+	"bytes"
+	"crypto/x509/pkix"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"storj.io/storj/internal/testcontext"
+	"storj.io/storj/internal/testidentity"
+	"storj.io/storj/internal/testpeertls"
+	"storj.io/storj/pkg/identity"
+	"storj.io/storj/pkg/peertls"
+	"storj.io/storj/pkg/peertls/extensions"
+	"storj.io/storj/storage"
+)
+
+func TestRevocationDB_Get(t *testing.T) {
+	ctx := testcontext.New(t)
+	defer ctx.Cleanup()
+
+	testidentity.RevocationDBsTest(ctx, t, func(t *testing.T, revDB extensions.RevocationDB, db storage.KeyValueStore) {
+		// NB: key indices are reversed as compared to chain indices
+		keys, chain, err := testpeertls.NewCertChain(2)
+		require.NoError(t, err)
+
+		ext, err := extensions.NewRevocationExt(keys[0], chain[peertls.LeafIndex])
+		require.NoError(t, err)
+
+		var rev *extensions.Revocation
+
+		{
+			t.Log("missing key")
+			rev, err = revDB.Get(chain)
+			assert.NoError(t, err)
+			assert.Nil(t, rev)
+
+			nodeID, err := identity.NodeIDFromKey(chain[peertls.CAIndex].PublicKey)
+			require.NoError(t, err)
+
+			err = db.Put(nodeID.Bytes(), ext.Value)
+			require.NoError(t, err)
+		}
+
+		{
+			t.Log("existing key")
+			rev, err = revDB.Get(chain)
+			assert.NoError(t, err)
+
+			revBytes, err := rev.Marshal()
+			assert.NoError(t, err)
+			assert.True(t, bytes.Equal(ext.Value, revBytes))
+		}
+	})
+}
+
+func TestRevocationDB_Put_success(t *testing.T) {
+	ctx := testcontext.New(t)
+	defer ctx.Cleanup()
+
+	testidentity.RevocationDBsTest(ctx, t, func(t *testing.T, revDB extensions.RevocationDB, db storage.KeyValueStore) {
+		// NB: key indices are reversed as compared to chain indices
+		keys, chain, err := testpeertls.NewCertChain(2)
+		require.NoError(t, err)
+
+		firstRevocation, err := extensions.NewRevocationExt(keys[0], chain[peertls.LeafIndex])
+		require.NoError(t, err)
+
+		time.Sleep(time.Second)
```

Comment:
why is this sleep here?

Reply:
I'm guessing based on the comment, it's because the timestamps need to be different so the time needs to change between creations. Is it possible instead of waiting until the time is actually different, can we just set the timestamp to be something else ?

Reply:
We _could_ pass the timestamp to `NewRevocationExt` but tests are the only place where we really want a time other than now.

Reply:
@egonelbre: what @coyle said